source/slang/slang-lower-to-ir.cpp

```diff
@@ -3592,6 +3592,41 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
         return newCaseLabel;
     }
 
+    bool hasSwitchCases(Stmt* inStmt)
+    {
+        Stmt* stmt = inStmt;
+        // Unwrap any surrounding `{ ... }` so we can look
+        // at the statement inside.
+        while (auto blockStmt = as<BlockStmt>(stmt))
+        {
+            stmt = blockStmt->body;
+            continue;
+        }
+
+        if (auto seqStmt = as<SeqStmt>(stmt))
+        {
+            // Walk through the children looking for cases
+            for (auto childStmt : seqStmt->stmts)
+            {
+                if (hasSwitchCases(childStmt))
```

Comment:
Minor point, but we should technically *not* recurse through inner/child `switch` statements, since their `case` or `default` statements don't count for the outer/parent statement.

Reply:
I'm just following the same logic as lowerSwitchCases, which seems to be finding cases that do belong to the switch?